src/db/db_setup.py

```diff
@@ -0,0 +1,79 @@
+import sqlite3
+from src.base.base_model import BaseModel
+from src.helpers.logger import LoggerFactory
+from config.settings import AppConfig
+
+logger = LoggerFactory.get_logger("AppDB")
+
+class SetupDB(BaseModel):
+    """
+    Database setup class that handles creation and initialization of the application's SQLite database.
+    
+    This class is responsible for creating the database file if it doesn't exist and
+    setting up all necessary tables using SQL scripts.
+    """
+
+    def __init__(self, app_config: AppConfig):
+        """
+        Initialize the SetupDB class.
+        
+        Args:
+            app_config (Config): Application configuration object
+        """
+        super().__init__(app_config)
+        self.config = app_config
+        self.db_path = self.config.db_path
+        self.funds_folder = self.config.funds_folder
+        self.sql_base_table_script_path = self.config.sql_query_base_tables
+        self.sql_master_reference_script_path = self.config.sql_query_master_reference
+
+    def _compute(self) -> None:
+        """
+        Execute the database setup process.
+        
+        This method creates the database if it doesn't exist and sets up tables
+        by executing the master reference and base table SQL scripts.
+        """
+        self.create_database()
+        self.setup_tables(self.sql_master_reference_script_path)
+        self.setup_tables(self.sql_base_table_script_path)
+
+    def create_database(self):
+        """
+        Create a new SQLite database if it doesn't exist.
+        
+        If the database already exists, this method will log this information and return
+        without making any changes.
+        """
+        if self.db_path.exists():
+            logger.info(f"Database already exists at {self.db_path}, skipping creation.")
```

Comment:
For this exercise, I would expect to use sqlite with an in-memory database. For production code, I would expect the database to be something other than sqlite (e.g. postgres).